packages/components/chip/src/Chip.styles.tsx

```diff
@@ -0,0 +1,183 @@
+import { makeVariants, tw } from '@spark-ui/internal-utils'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+import { dashedVariants, filledVariants, outlinedVariants, tintedVariants } from './variants'
+
+export const chipStyles = cva(
+  [
+    'gap-md text-body-1 h-sz-32 font-regular box-border inline-block rounded-md',
+    'focus-visible:ring-outline-high ring-inset focus-visible:outline-none focus-visible:ring-2',
+  ],
+  {
+    variants: {
+      /**
+       * Main style of the chip.
+       *
+       * - `filled`: Chip will be plain.
+       *
+       * - `outlined`: Chip will be transparent with an outline.
+       *
+       * - `tinted`: Chip will be filled but using a lighter color scheme.
+       *
+       * - `dashed`: Chip will be transparent with an outline dashed.
+       */
+      design: makeVariants<'design', ['filled', 'outlined', 'tinted', 'dashed']>({
+        filled: [],
+        outlined: ['bg-transparent', 'ring-1', 'ring-current'],
+        tinted: [],
+        dashed: [
+          'bg-transparent border-sm border-dashed shadow-none focus-visible:border-outline-high',
+        ],
+      }),
+      /**
+       * Color scheme of the chip.
+       */
+      intent: makeVariants<
+        'intent',
+        ['primary', 'secondary', 'success', 'alert', 'danger', 'info', 'neutral', 'surface']
+      >({
+        primary: [],
+        secondary: [],
+        success: [],
+        alert: [],
+        danger: [],
+        info: [],
+        neutral: [],
+        surface: [],
+      }),
+      /**
+       * Disable the chip, preventing user interaction and adding opacity.
+       */
+      disabled: {
+        true: ['cursor-not-allowed', 'opacity-dim-3'],
+      },
+    },
+    compoundVariants: [
+      ...filledVariants,
+      ...outlinedVariants,
+      ...tintedVariants,
+      ...dashedVariants,
+    ],
+    defaultVariants: {
+      design: 'outlined',
+      intent: 'primary',
+    },
+  }
+)
+
+export const chipContentStyles = cva(
+  ['gap-sm inline-flex h-full items-center justify-center whitespace-nowrap'],
+  {
+    variants: {
+      isBordered: {
+        no: ['max-w-[240px]'],
+        yes: ['max-w-[238px]'],
+      },
+      mode: {
+        default: [],
+        icon: [],
+      },
+      hasCloseIcon: {
+        yes: [],
+        no: [],
+      },
+    },
+    defaultVariants: {
+      isBordered: 'no',
+      mode: 'default',
+      hasCloseIcon: 'no',
+    },
+    compoundVariants: [
+      {
+        mode: 'default',
+        hasCloseIcon: 'no',
+        isBordered: 'no',
+        class: tw(['px-md']),
+      },
+      {
+        mode: 'default',
+        hasCloseIcon: 'no',
+        isBordered: 'yes',
+        class: tw(['px-[7px]']),
```

Comment:
Okay :) @andresz1 had a similar issue on the textarea PR, maybe for those cases we could use `calc()` to use the appropriate token minus a delta ? It will be more scalable that way.